go-controller/pkg/kubevirt/pod.go

```diff
@@ -0,0 +1,213 @@
+
+	for _, vmPod := range vmPods {
+		podAnnotation, err := util.UnmarshalPodAnnotation(vmPod.Annotations, nadName)
+		if err == nil {
+			return podAnnotation, nil
+		}
+	}
+	return nil, fmt.Errorf("missing virtual machine pod annotation at stale pods for %s/%s", pod.Namespace, pod.Name)
+}
+
+// EnsurePodAnnotationForVM will at live migration extract the ovn pod
+// annotations from the source vm pod and copy it
+// to the target vm pod so ip address follow vm during migration. This has to
+// done before creating the LSP to be sure that Address field get configured
+// correctly at the target VM pod LSP.
+func EnsurePodAnnotationForVM(watchFactory *factory.WatchFactory, kube *kube.KubeOVN, pod *corev1.Pod, netInfo util.NetInfo, nadName string) (*util.PodAnnotation, error) {
+	if !IsPodLiveMigratable(pod) {
+		return nil, nil
+	}
+
+	if podAnnotation, err := util.UnmarshalPodAnnotation(pod.Annotations, nadName); err == nil {
+		return podAnnotation, nil
+	}
+
+	podAnnotation, err := findPodAnnotation(watchFactory, pod, netInfo, nadName)
+	if err != nil {
+		return nil, err
+	}
+
+	var modifiedPod *corev1.Pod
+	resultErr := retry.RetryOnConflict(util.OvnConflictBackoff, func() error {
+		// Informer cache should not be mutated, so get a copy of the object
+		pod, err := watchFactory.GetPod(pod.Namespace, pod.Name)
+		if err != nil {
+			return err
+		}
+		// Informer cache should not be mutated, so get a copy of the object
+		modifiedPod = pod.DeepCopy()
+		if podAnnotation != nil {
+			modifiedPod.Annotations, err = util.MarshalPodAnnotation(modifiedPod.Annotations, podAnnotation, nadName)
+			if err != nil {
+				return err
+			}
+		}
+		return kube.UpdatePod(modifiedPod)
+	})
+	if resultErr != nil {
+		return nil, fmt.Errorf("failed to update labels and annotations on pod %s/%s: %v", pod.Namespace, pod.Name, resultErr)
+	}
+	return podAnnotation, nil
+}
+
+// IsMigratedSourcePodStale return true if there are other pods related to
+// to it and any of them has newer creation timestamp.
+func IsMigratedSourcePodStale(client *factory.WatchFactory, pod *corev1.Pod) (bool, error) {
+	vmPods, err := findVMRelatedPods(client, pod)
+	if err != nil {
+		return false, fmt.Errorf("failed finding related pods for pod %s/%s when checking live migration left overs: %v", pod.Namespace, pod.Name, err)
+	}
+
+	for _, vmPod := range vmPods {
+		if vmPod.CreationTimestamp.After(pod.CreationTimestamp.Time) {
+			return true, nil
+		}
+	}
+
+	return false, nil
+}
+
+// ExtractVMNameFromPod retunes namespace and name of vm backed up but the pod
+// for regular pods return nil
+func ExtractVMNameFromPod(pod *corev1.Pod) *ktypes.NamespacedName {
+	vmName, ok := pod.Labels[kubevirtv1.VirtualMachineNameLabel]
+	if !ok {
+		return nil
+	}
+	return &ktypes.NamespacedName{Namespace: pod.Namespace, Name: vmName}
+}
+
+func CleanUpForVM(controllerName string, nbClient libovsdbclient.Client, watchFactory *factory.WatchFactory, pod *corev1.Pod, networkName string) error {
```

Comment:
nit: `CleanUpLiveMigratablePod`?

Reply:
I would prefer `CleanUpLiveMigratableVM` - a pod doesn't migrate.

Reply:
We are using the term `LiveMigratablePod` all over the place, but I am ok with that other name as well

Reply:
going with `CleanUpLiveMigratablePod` since we are passing the pod